cmd/ctl/pkg/install/install.go

```diff
@@ -116,6 +142,8 @@ func NewCmdInstall(ctx context.Context, ioStreams genericclioptions.IOStreams) *
 	}
 
 	settings.AddFlags(cmd.Flags())
+	// Here we set the default value shown in the usage message.
+	// The actual default value is overridden in the RunE function above.
 	cmd.Flag("namespace").DefValue = defaultCertManagerNamespace
```

Comment:
```suggestion
	settings.AddFlags(cmd.Flags())

	// Here we set the default value shown in the usage message.
	cmd.Flag("namespace").DefValue = defaultCertManagerNamespace
	// Here we set the default value.
	cmd.Flag("namespace").Value.Set(defaultCertManagerNamespace)
```

This should give the same result (overwriting the default value for the namespace flag).

Reply:
I did try that, but then noticed that `Value.Set` returns an `err` so I moved it to the Run function where I had somewhere to return the error.

Reply:
My guess is it was likely this PR https://github.com/jetstack/cert-manager/pull/4407

Reply:
Seems like the err is always nil for string flags:
https://github.com/spf13/pflag/blob/85dd5c8bc61cfa382fecd072378089d4e856579d/string.go#L11-L14

Originally, the code used the shared factory & flags. I think it is better to not do that here, because the Helm libs have their own set of Flags.

Reply:
Can we just check for an error & print + exit if it happens & add a comment to indicate it should never happen?

Reply:
Ok. If the err is always nil, then I'll just drop it, as you originally suggested.